packages/transaction-controller/src/TransactionController.ts

```diff
@@ -306,6 +333,20 @@ export class TransactionController extends BaseController<
       onNetworkStateChange: (listener: (state: NetworkState) => void) => void;
       provider: Provider;
       securityProviderRequest?: SecurityProviderRequest;
+      hooks: {
+        afterSign?: (
+          transactionMeta: TransactionMeta,
+          signedTx: TypedTransaction,
+        ) => boolean;
+        beforeApproveOnInit?: (transactionMeta: TransactionMeta) => boolean;
```

Comment:
Should we be using this?

Reply:
This one is waiting for Goktug's [PR](https://github.com/MetaMask/core/pull/1916/files#r1374302451). I will need to rebase and properly use it.